### PR TITLE
switch to ml_configuration in agent response as backend changed

### PR DESCRIPTION
**Pull request by qianheng-aws**
### Description
The backend ml-commons changed its configuration field from `configuration` to `ml_configuration` and lead to error.

```
GET /_plugins/_ml/config/os_chat

{
  "config_type": "os_chat_root_agent",
  "ml_configuration": {
    "agent_id": "******************"
  }
}
```

Related change in ml-commons:
https://github.com/opensearch-project/ml-commons/pull/2710

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test.
- [ ] New functionality has user manual doc added.
- [ ] Commits are signed per the DCO using --signoff.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Hailong-am — APPROVED** [review]
would you mind add changelog  for this change?

File: server/routes/get_agent.ts [hunk 1]
```diff
@@ -14,10 +14,10 @@ export const getAgent = async (id: string, client: OpenSearchClient['transport']
       path,
     });
 
-    if (!response || !response.body.configuration?.agent_id) {
+    if (!response || !response.body.ml_configuration?.agent_id) {
       throw new Error(`cannot get agent ${id} by calling the api: ${path}`);
     }
-    return response.body.configuration.agent_id;
+    return response.body.ml_configuration.agent_id;
```
Comment (Hailong-am) [inline server/routes/get_agent.ts:20]: maybe add a fallback here to make it compatible with existing data
```suggestion
    return response.body.ml_configuration.agent_id || response.body. configuration.agent_id;
```

**Comment by wanglam** [comment]
Is this change target for 2.16?

File: server/routes/get_agent.ts (hunk 1 above)
Comment (wanglam) [reply server/routes/get_agent.ts:20]: The code will throw error if `!response.body.ml_configuration?.agent_id` above. I think `response.body. configuration.agent_id` won't be executed. Need to find some way to compatible with legacy data.

**Comment by qianheng-aws** [comment]
> Is this change target for 2.16?

The ml-commons related change https://github.com/opensearch-project/dashboards-assistant/pull/239#issuecomment-2247335910 has backported to 2.16, so I think this change should target for 2.16

File: server/routes/get_agent.ts (hunk 1 above)
Comment (Hailong-am) [reply server/routes/get_agent.ts:20]: yes, the if condition should change accordingly.
```typescript
if (!response || !response.body.ml_configuration?.agent_id || !response.body.configuration?.agent_id) {
      throw new Error(`cannot get agent ${id} by calling the api: ${path}`);
    }
```

Comment (qianheng-aws) [reply server/routes/get_agent.ts]: I changed the condition of throwing error as well.

File: server/routes/get_agent.ts [hunk 2]
```diff
@@ -14,10 +14,13 @@ export const getAgent = async (id: string, client: OpenSearchClient['transport']
...
 
-    if (!response || !response.body.ml_configuration?.agent_id) {
+    if (
+      !response ||
+      !(response.body.ml_configuration?.agent_id || response.body.configuration?.agent_id)
+    ) {
       throw new Error(`cannot get agent ${id} by calling the api: ${path}`);
     }
-    return response.body.ml_configuration.agent_id;
+    return response.body.ml_configuration.agent_id || response.body.configuration.agent_id;
```
Comment (wanglam) [inline server/routes/get_agent.ts:23]: I'd prefer `response.body.ml_configuration?.agent_id || response.body.configuration.agent_id`, `body.ml_configuration` may undefined right?

File: server/routes/get_agent.ts (hunk 1 above)
Comment (qianheng-aws) [reply server/routes/get_agent.ts]: @Hailong-am  Maybe && between the configuration since either agent_id can be retrieved will be ok

**Review by wanglam — APPROVED** [review]
LGTM

**Review by raintygao — APPROVED** [review]

**Comment by raintygao** [comment]
It would be better to add your change in https://github.com/opensearch-project/dashboards-assistant/blob/main/release-notes/dashboards-assistant.release-notes-2.16.0.0.md as this should be released in 2.16.

**Comment by qianheng-aws** [comment]
> It would be better to add your change in https://github.com/opensearch-project/dashboards-assistant/blob/main/release-notes/dashboards-assistant.release-notes-2.16.0.0.md as this should be released in 2.16.

@raintygao  But it cannot be called a feature nor bug fix. Maybe `Maintenance `?

**Comment by raintygao** [comment]
> > It would be better to add your change in https://github.com/opensearch-project/dashboards-assistant/blob/main/release-notes/dashboards-assistant.release-notes-2.16.0.0.md as this should be released in 2.16.
> 
> @raintygao But it cannot be called a feature nor bug fix. Maybe `Maintenance `?

Im okay to this.

**Pull request merged**